plugins/callback/mail.py

```diff
@@ -94,34 +95,36 @@ def set_options(self, task_keys=None, var_options=None, direct=None):
         self.cc = self.get_option('cc')
         self.bcc = self.get_option('bcc')
 
+    @staticmethod
+    def encode_addresses(addresses):
+        addresses_list = addresses.split(',')
+        addresses_list = map(lambda x: email.utils.formataddr(email.utils.parseaddr(x), charset='utf-8'), addresses_list)
+        return ','.join(addresses_list)
+
     def mail(self, subject='Ansible error mail', body=None):
         if body is None:
             body = subject
 
         smtp = smtplib.SMTP(self.smtphost, port=self.smtpport)
 
-        b_sender = to_bytes(self.sender)
-        b_to = to_bytes(self.to)
-        b_cc = to_bytes(self.cc)
-        b_bcc = to_bytes(self.bcc)
-        b_subject = to_bytes(subject)
-        b_body = to_bytes(body)
-
-        b_content = b'From: %s\n' % b_sender
-        b_content += b'To: %s\n' % b_to
+        content = 'From: %s\n' % self.encode_addresses(self.sender)
+        content += 'Message-ID: %s\n' % email.utils.make_msgid()
+        content += 'To: %s\n' % self.encode_addresses(self.to)
         if self.cc:
-            b_content += b'Cc: %s\n' % b_cc
-        b_content += b'Subject: %s\n\n' % b_subject
-        b_content += b_body
+            content += 'Cc: %s\n' % self.encode_addresses(self.cc)
+        content += 'Subject: %s\n' % subject.strip()
+        content += 'Date: %s\n' % email.utils.formatdate()
+        content += 'Content-Type: text/plain; charset=utf-8\n'
```

Comment:
```suggestion
        content += 'Content-Type: text/plain; charset=utf-8\n\n'
```
The double newline (which was present in the original code) is needed to separate headers from body.

Reply:
Thanks !
This is now fixed.